esdm/esdm_drng_mgr.c

```diff
@@ -772,7 +776,7 @@ static ssize_t esdm_drng_get(struct esdm_drng *drng, uint8_t *outbuf,
 			/* Do not produce more than DRNG security strength. */
 			todo = min_uint32(todo, esdm_security_strength() >> 3);
 
-			atomic_set(&drng->internal_bits, 0);
+			atomic_set(&drng->request_bits_since_fully_seeded, 0);
```

Comment:
This code is still here - shouldn't it be removed? Or what is the reason for the code? IMHO this is no place to reset the counter.

Reply:
Where's the right or a better place for this? In `esdm_drng_seed_es_nolock` after the while loop and before `return collected_entropy;`?

Reply:
I am not sure we need it at all. It value should be reset to 0 when there is a reseed. This is already in esdm_drng_inject that does the actual reseed.

Reply:
I tested without this and the bit count will grow above both limits. But I think you are right, that this is not needed in every inject, but only after a full reseed. I therefore moved it to `esdm_drng_seed_es_nolock`.